fhir-server/src/main/java/au/csiro/pathling/fhirpath/encoding/QuantityEncoding.java

```diff
@@ -0,0 +1,98 @@
+      quantity.setValue(
+          quantity.getValue().setScale(DecimalCustomCoder.scale(), RoundingMode.HALF_UP));
+    }
+    return RowFactory.create(quantity.getId(), quantity.getValue(), quantity.getValue().scale(),
+        comparator, quantity.getUnit(), quantity.getSystem(), quantity.getCode(), null /* _fid */);
+  }
+
+  @Nonnull
+  public static Quantity decode(@Nonnull final Row row) {
+    final Quantity quantity = new Quantity();
+
+    Optional.ofNullable(row.getString(0)).ifPresent(quantity::setId);
+
+    // The value gets converted to a BigDecimal, taking into account the scale that has been encoded 
+    // alongside it.
+    final int scale = row.getInt(2);
+    final BigDecimal value = Optional.ofNullable(row.getDecimal(1))
+        .map(bd -> bd.scale() > DecimalCustomCoder.scale()
+                   ? bd.setScale(scale, RoundingMode.HALF_UP)
+                   : bd)
+        .orElse(null);
+    quantity.setValue(value);
+
+    // The comparator is encoded as a string code, we need to convert it back to an enum.
+    Optional.ofNullable(row.getString(3))
+        .map(QuantityComparator::fromCode)
+        .ifPresent(quantity::setComparator);
+
+    Optional.ofNullable(row.getString(4)).ifPresent(quantity::setUnit);
+    Optional.ofNullable(row.getString(5)).ifPresent(quantity::setSystem);
+    Optional.ofNullable(row.getString(6)).ifPresent(quantity::setCode);
+
+    return quantity;
+  }
+
+  @Nonnull
+  public static StructType dataType() {
+    final Metadata metadata = new MetadataBuilder().build();
+    final StructField id = new StructField("id", DataTypes.StringType, true, metadata);
+    final StructField value = new StructField("value", DataTypes.createDecimalType(
+        DecimalCustomCoder.precision(), DecimalCustomCoder.scale()), true, metadata);
+    final StructField valueScale = new StructField("value_scale", DataTypes.IntegerType, true,
+        metadata);
+    final StructField comparator = new StructField("comparator", DataTypes.StringType, true,
+        metadata);
+    final StructField unit = new StructField("unit", DataTypes.StringType, true, metadata);
+    final StructField system = new StructField("system", DataTypes.StringType, true, metadata);
+    final StructField code = new StructField("code", DataTypes.StringType, true, metadata);
+    final StructField canonicalizedValue = new StructField(CANONICALIZED_VALUE_COLUMN,
+        DataTypes.createDecimalType(
+            DecimalCustomCoder.precision(), DecimalCustomCoder.scale()), true, metadata);
+    final StructField canonicalizedCode = new StructField(CANONICALIZED_CODE_COLUMN,
+        DataTypes.StringType, true, metadata);
+    final StructField fid = new StructField("_fid", DataTypes.IntegerType, true,
+        metadata);
+    return new StructType(
+        new StructField[]{id, value, valueScale, comparator, unit, system, code, canonicalizedValue,
+            canonicalizedCode, fid});
+  }
+
```

Comment:
This (and other encoding like this, e.g. for SimpleCoding) should be placed closer to the `encoders` module.
Minimally the encoders should provide an interface that encapsulates encode/decode/datatype operations (possibly with the current implementation).
Ideally we should be able to use the builders/converters in encoders to provide the implementation.

Reply:
Captured as #868